src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -1249,6 +1345,57 @@ std::string ImageReplayer<I>::to_string(const State state) {
 }
 
 template <typename I>
+void ImageReplayer<I>::resync_image(Context *on_finish) {
+  dout(20) << dendl;
+
+  bool is_stopped = false;
+  {
+    Mutex::Locker l(m_lock);
+
+    m_stopping_for_resync = true;
+    if (m_state == STATE_STOPPED) {
```

Comment:
I'm not finding a simple way of hooking the asok method to librbd's request resync. Only through a new method in ImageReplayer as I need to access either to the local image context, or to the local cluster connection and the local image id/name.
What if we completely remove the resync command from the asok?